tests/test_app.py

```diff
@@ -1,131 +1,79 @@
-                "idList": "604e8d8f40bbb0669d50314e",
-                "dueComplete": False,
-                "due": None,
-            },
-        ]
-        return response
-    elif url == f'https://api.trello.com/1/boards/{TEST_BOARD_ID}/lists':
-        response = Mock()
-        response.json.return_value = [
-            {
-                "id": "604e8d8f40bbb0669d50314e",
-                "name": "To Do",
-                "idBoard": "0123456789",
-                "subscribed": False
-            },
-            {
-                "id": "DoneID",
-                "name": "Done",
-                "idBoard": "0123456789",
-                "subscribed": False
-            },
-            {
-                "id": "DoingID",
-                "name": "Doing",
-                "idBoard": "0123456789",
-                "subscribed": False
-            },
-        ]
-        return response
-    return None
-
-
 @pytest.fixture
 def client():
     # Use our test integration config instead of the 'real' version
     file_path = find_dotenv('.env.test')
     load_dotenv(file_path, override=True)
 
-    # Create the new app
-    with patch('requests.get') as mock_get_requests:
-        mock_get_requests.side_effect = mock_get_response
+    with mongomock.patch(servers=(('fakemongo.com', 27017),)):
+        prepare_test_data()
         test_app = app.create_app()
+        with test_app.test_client() as client:
+            yield client
 
-    # Use the app to create a test_client that can be used in our tests
-    with test_app.test_client() as client:
-        yield client
 
-
-@patch('requests.get')  
-def test_index_page(mock_get_requests, client):
-    mock_get_requests.side_effect = mock_get_response
+def test_index_page(client):
     response = client.get('/')
     response_text = response.get_data(as_text=True)
+    assert 'Do Corndel project exercises!' in response_text
     assert 'No recent items completed yet' in response_text 
     assert 'No items in progress' in response_text 
 
 
-@patch('requests.post')  
-@patch('requests.get')  
-def test_add_todo_item(mock_get_requests, mock_post_requests, client):
-    mock_post_requests.side_effect = None 
-    mock_get_requests.side_effect = mock_get_response
-
+def test_add_todo_item(client):
     response = client.post('/add_item', data=dict(
-        title='name',
+        title='Do laundry',
         description='description',
     ), follow_redirects=True)
     response_text = response.get_data(as_text=True)
     assert 'No recent items completed yet' in response_text 
     assert 'No items in progress' in response_text 
+    assert 'Do laundry' in response_text 
 
 
-@patch('requests.get')  
-def test_complete_item(mock_get_requests, client):
-    mock_get_requests.side_effect = mock_get_response
-
-    response = client.get('/complete/1', follow_redirects=True)
+def test_complete_item(client):
+    response = client.get(f'/complete/61a7fab58503e36eb7595414', follow_redirects=True)
```

Comment:
I don't think these tests are really doing anything, since I can't see anywhere this id matches - meaning that the endpoint just reports success even though nothing has been changed. Is that correct?

For example, your complete test asserts `assert 'No recent items completed yet' in response_text` which seems wrong, or your test for marking an item in progress saying `assert 'No items in progress' in response_text`!

It would be good to make sure they are actually acting on the relevant item, so e.g. add the relevant id field to the item you add in your test data:
```python
    item = {
        "_id": ObjectId("61a7fab58503e36eb7595414"),
        "name": "Do Corndel project exercises!",
        ....
```

And then having sensible asserts for each method, e.g. after completing you want to check:
```python
    assert 'No items to do yet' in response_text
    assert 'No items in progress' in response_text 
    assert 'No recent items completed yet' not in response_text 
```

Reply:
Very well spotted. I totally forgot to correct that